### ISSUE TITLE
Bug when compiling on Windows 11 platform 

### ISSUE THREAD

**Issue by radiantmediaplayer** [shaka-project/shaka-player#4817]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Y

**What version of Shaka Player are you using?**
4.3.1

**Can you reproduce the issue with our latest release version?**
Y

**Can you reproduce the issue with the latest code from `main`?**
Y

**Are you using the demo app or your own custom app?**
N/A

**If custom app, can you reproduce the issue using our demo app?**
N/A

**What browser and OS are you using?**
N/A

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
N/A

**What did you do?**
I am on Windows 11, version 22H2, x64
I am building Shaka player from source with command line:
`python build/build.py +@networking +@manifests +@text +./node_modules/eme-encryption-scheme-polyfill/index.js +./lib/polyfill/encryption_scheme.js +./lib/polyfill/mediasource.js +./lib/polyfill/media_capabilities.js +./lib/polyfill/patchedmediakeys_nop.js +./lib/polyfill/patchedmediakeys_webkit.js  +./lib/polyfill/random_uuid.js +./lib/polyfill/storage_estimate.js +./lib/polyfill/symbol.js +./lib/polyfill/videoplaybackquality.js +./lib/polyfill/vttcue.js  --debug`

This works fine with 4.3.0, but when I switch to 4.3.1 branch build process fails with error: 
```
Traceback (most recent call last):
  File "C:\wamp64\www\shaka-player\build\build.py", line 403, in <module>
    shakaBuildHelpers.run_main(main)
  File "C:\wamp64\www\shaka-player\build\shakaBuildHelpers.py", line 350, in run_main
    sys.exit(main(sys.argv[1:]))
  File "C:\wamp64\www\shaka-player\build\build.py", line 396, in main
    if not custom_build.build_library(name, locales, force, is_debug):
  File "C:\wamp64\www\shaka-player\build\build.py", line 302, in build_library
    if not closure.compile(closure_opts, force):
  File "C:\wamp64\www\shaka-player\build\compiler.py", line 144, in compile
    output_options += self._prepare_wrapper()
  File "C:\wamp64\www\shaka-player\build\compiler.py", line 189, in _prepare_wrapper
    raise RuntimeError('Failed to strip whitespace from wrapper!')
RuntimeError: Failed to strip whitespace from wrapper!
```

Can you look into it?

**Comment by joeyparrish** [comment]
Please run `npm ci` to install node dependencies.  The step that failed involves the Closure Compiler, which is installed into `node_modules`.

Does this help?

**Comment by joeyparrish** [comment]
If that doesn't resolve your issue, please provide the full console output, as there may be more information above the python traceback that points to the real cause.

**Comment by radiantmediaplayer** [comment]
`npm ci` does not solve the issue. 
If I switch to 4.3.0 release it compiles without issue. If I go for 4.3.1 or 4.3.2 then the error appears and build process fails. The full console output is below. 
I can build on Linux Ubuntu 4.3.1 or 4.3.2 so it seems a build error was introduced in 4.3.1 for Windows. 

`python build/build.py +@networking +@manifests +@text +@lcevc +./node_modules/eme-encryption-scheme-polyfill/index.js +./lib/polyfill/encryption_scheme.js +./lib/polyfill/mediasource.js +./lib/polyfill/media_capabilities.js +./lib/polyfill/patchedmediakeys_nop.js +./lib/polyfill/patchedmediakeys_webkit.js  +./lib/polyfill/random_uuid.js +./lib/polyfill/storage_estimate.js +./lib/polyfill/symbol.js +./lib/polyfill/videoplaybackquality.js +./lib/polyfill/vttcue.js  --debug`
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated @babel/polyfill@7.12.1: 🚨 This package has been deprecated in favor of separate inclusion of a polyfill and regenerator-runtime (when needed). See the @babel/polyfill docs (https://babeljs.io/docs/en/babel-polyfill) for more information.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.0: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated open-sans-fonts@1.6.2: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
npm WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
[INFO] Compiling the library (ui, debug)...
Traceback (most recent call last):
  File "C:\wamp64\www\shaka-player\build\build.py", line 403, in <module>
    shakaBuildHelpers.run_main(main)
  File "C:\wamp64\www\shaka-player\build\shakaBuildHelpers.py", line 350, in run_main
    sys.exit(main(sys.argv[1:]))
  File "C:\wamp64\www\shaka-player\build\build.py", line 396, in main
    if not custom_build.build_library(name, locales, force, is_debug):
  File "C:\wamp64\www\shaka-player\build\build.py", line 302, in build_library
    if not closure.compile(closure_opts, force):
  File "C:\wamp64\www\shaka-player\build\compiler.py", line 144, in compile
    output_options += self._prepare_wrapper()
  File "C:\wamp64\www\shaka-player\build\compiler.py", line 189, in _prepare_wrapper
    raise RuntimeError('Failed to strip whitespace from wrapper!')
RuntimeError: Failed to strip whitespace from wrapper!

**Comment by joeyparrish** [comment]
What shell are you using on Windows?  PowerShell?  CMD?  Git Bash?  Cygwin?

Also, can you please run the build command with the environment variable PRINT_ARGUMENTS=1 set?  That should show us the exact command that is failing to run.

**Comment by radiantmediaplayer** [comment]
I am using Git Bash. I did add PRINT_ARGUMENTS=1 to my Windows system environment variable. 
It printed an error about JAVA class v55 being needed to compile so I updated my JAVA set up and now I can compile like before without issue. This is weird though that this error would pop up from 4.3.0 to 4.3.1.
I used to have JAVA 8 update 311 + JDK 8 Update 281. I now have JAVA 8 Update 351 and JDK 19.0.1 and it works. I needed JDK 8 for another legacy project but that is a different story. I am closing this, thanks for your help.
